models/userModel.js

```diff
@@ -0,0 +1,52 @@
+const mongoose = require("mongoose");
+const validator = require("validator");
+const bcrypt = require("bcrypt");
+const userSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: [true, "Pls, tell us your name"],
+  },
+  email: {
+    type: String,
+    required: [true, "Please provide your email"],
+    unique: true,
+    lowercase: true,
+    validate: [validator.isEmail, "Please provide a valid email"],
+  },
+  photo: {
+    type: String,
+  },
+  password: {
+    type: String,
+    required: [true, "Please provide a password"],
+    minLength: 8,
+    select: false,
+  },
+  passwordConfirm: {
```

Comment:
You don't store confirm password in database

Reply:
i'm not saving it, on save, i'm setting it to undefined

Reply:
It should never be in your model. Either you are saving it or not, it's should not be among the model attributes